modules/swanBidAdapter.js

```diff
@@ -0,0 +1,263 @@
+import {config} from '../src/config.js';
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import * as utils from '../src/utils.js';
+import {BANNER, VIDEO} from '../src/mediaTypes.js';
+import includes from 'core-js-pure/features/array/includes.js';
+
+const bidderConfig = 'hb_pb_ortb';
+const bidderVersion = '1.0';
+const VIDEO_TARGETING = ['startdelay', 'mimes', 'minduration', 'maxduration',
+  'startdelay', 'skippable', 'playbackmethod', 'api', 'protocols', 'boxingallowed',
+  'linearity', 'delivery', 'protocol', 'placement', 'minbitrate', 'maxbitrate', 'ext'];
+
+export const spec = {
+  code: 'swan',
+  supportedMediaTypes: [BANNER, VIDEO],
+  isBidRequestValid,
+  buildRequests,
+  interpretResponse,
+  getUserSyncs,
+  transformBidParams
+};
+
+registerBidder(spec);
+
+function transformBidParams(params, isOpenRtb) {
+  return utils.convertTypes({
+    'unit': 'string',
+    'customFloor': 'number'
+  }, params);
+}
+
+function isBidRequestValid(bidRequest) {
+  return !!(bidRequest.params.swan_offer_name && bidRequest.params.endpoint);
+}
+
+function buildRequests(bids, bidderRequest) {
+  let videoBids = bids.filter(bid => isVideoBid(bid));
+  let bannerBids = bids.filter(bid => isBannerBid(bid));
+  let requests = bannerBids.length ? [createBannerRequest(bannerBids, bidderRequest)] : [];
+  videoBids.forEach(bid => {
+    requests.push(createVideoRequest(bid, bidderRequest));
+  });
+  return requests;
+}
+
+function createBannerRequest(bids, bidderRequest) {
+  let data = getBaseRequest(bids[0], bidderRequest);
+  data.imp = bids.map(bid => {
+    const floor = getFloor(bid, BANNER);
+    let imp = {
+      id: bid.bidId,
+      tagid: bid.adUnitCode,
+      banner: {
+        format: toFormat(bid.mediaTypes.banner.sizes),
+        topframe: utils.inIframe() ? 0 : 1
+      },
+      ext: {
+        Offername: bid.params.swan_offer_name,
+        Offer: window.swan_offers[bid.params.swan_offer_name]
```

Comment:
bidders who are given the swan offer name are able to retrieve and pass the offer to themselves